### PR TITLE
Add opreator AggregateActivityDatesOperator

### PR DESCRIPTION
**Pull request by abalhomaid**
part of intern refactored code

### CODE REVIEW

File: tasrif/processing_pipeline/custom/change_date_format_to_start_to_end.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    started the sleep activity and when the participant ended that activity. This operator assumes that
+    `date_feature_name` is sorted.
+
+    The pseudo code of this operator is to group by participantIdentifier, fetch the first date, and end date
+    of each group, then return a list of [participantIdentifier, startDate, endDate].
+    See example below for clarification.
+
+    >>> import pandas as pd
+    >>>
+    >>> from tasrif.processing_pipeline.custom import ChangeDateFormatToStartToEnd
```
Comment (uabbas) [inline tasrif/processing_pipeline/custom/change_date_format_to_start_to_end.py:24]: I think the name of the class `ChangeDateFormatToStartToEnd` is a bit misleading. It is actually aggregating the rows in a special way such that the start and end time per record is known. `AggregateActivityDatesOperator` or something similar would be more appropriate.

File: tasrif/processing_pipeline/custom/change_date_format_to_start_to_end.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    2   144     1   2016-03-13 06:36:00     2016-03-13 06:45:00
+    3   167     1   2016-03-14 01:32:00     2016-03-14 01:41:00
+    4   167     2   2016-03-15 02:36:00     2016-03-15 02:45:00
+    5   167     3   2016-03-15 03:03:00     2016-03-15 03:12:00
+    6   167     4   2016-03-15 03:58:00     2016-03-15 04:07:00
+    """
+    def __init__(self, # pylint: disable=R0913
+                 date_feature_name,
+                 participant_identifier,
+                 aggregation_definition=None,
```
Comment (uabbas) [inline tasrif/processing_pipeline/custom/change_date_format_to_start_to_end.py:115]: Do we really need aggregation_defintition toe be passed here?

File: tasrif/processing_pipeline/custom/change_date_format_to_start_to_end.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+                 start_column_name='start',
+                 end_column_name='end'):
+        """Creates a new instance of EncodeCyclicalFeaturesOperator
```
Comment (uabbas) [inline tasrif/processing_pipeline/custom/change_date_format_to_start_to_end.py:118]: class name is not `EncodeCyclicalFeaturesOperator`

**Review by uabbas — CHANGES_REQUESTED** [review]
Review comments are marked inline with code.

File: tasrif/processing_pipeline/custom/change_date_format_to_start_to_end.py (hunk 1 above)
Comment (abalhomaid) [reply tasrif/processing_pipeline/custom/change_date_format_to_start_to_end.py:24]: Thanks Ummar, I will address the comments during the upcoming weekend

File: tasrif/processing_pipeline/custom/change_date_format_to_start_to_end.py (hunk 2 above)
Comment (abalhomaid) [reply tasrif/processing_pipeline/custom/change_date_format_to_start_to_end.py:115]: Good question. `aggregation_definition` parameter gives the user the ability to aggregate non-date columns.
In the example, the aggregation occurs in the `value` column
by summing the values and summing only the value of ones as shown below

`{'value': [np.sum, lambda x: x[x == 1].sum()]}`

```

File: tasrif/processing_pipeline/custom/change_date_format_to_start_to_end.py (hunk 1 above)
Comment (abalhomaid) [reply tasrif/processing_pipeline/custom/change_date_format_to_start_to_end.py]: Updated the repo accordingly

File: tasrif/processing_pipeline/custom/aggregate_activity_dates_operator.py [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+    5   167     3   2016-03-15 03:03:00     2016-03-15 03:12:00
+    6   167     4   2016-03-15 03:58:00     2016-03-15 04:07:00
+    """
+    def __init__(self, # pylint: disable=R0913
+                 date_feature_name,
+                 participant_identifier,
+                 aggregation_definition=None,
+                 start_column_name='start',
+                 end_column_name='end'):
+        """Creates a new instance of ChangeDateFormatToStartToEnd
```
Comment (uabbas) [inline tasrif/processing_pipeline/custom/aggregate_activity_dates_operator.py:120]: Class name needs to be corrected.

**Review by uabbas — CHANGES_REQUESTED** [review]

**Review by uabbas — APPROVED** [review]

**Pull request merged**